src/Tempest/Database/QueryStatement.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database;
+
+use RuntimeException;
+use Stringable;
+use Tempest\Database\Drivers\MySqlDriver;
+use Tempest\Database\Drivers\PostgreSqlDriver;
+use Tempest\Database\Drivers\SQLiteDriver;
+use UnhandledMatchError;
+
+final class QueryStatement implements Stringable
+{
+    private array                   $query = [];
+
+    public function __construct(
+        private readonly DatabaseDriver $driver,
+        private string $table,
+    ) {
+    }
+
+    public function create(callable $callback): self
+    {
+        if (! empty($this->query)) {
+            throw new RuntimeException('create statement should be the first statement');
+        }
+
+        $this->query[] = sprintf(
+            "CREATE TABLE %s (%s);",
+            $this->table,
+            $callback($this->driver->createQueryStatement($this->table))
+        );
+
+        return $this;
+    }
+
+    /** @throws UnhandledMatchError */
+    public function alterTable(string $action, callable $callback): self
+    {
+        if (! empty($this->query)) {
+            throw new RuntimeException('alter statement should be the first statement');
+        }
+
+        $operation = match ($this->driver::class) {
+            MySqlDriver::class => sprintf('%s', strtoupper($action)),
+            PostgreSqlDriver::class,
+            SQLiteDriver::class => sprintf('%s COLUMN', strtoupper($action)),
+        };
+
+        $this->query[] = sprintf( // @coverage
```

Comment:
What does `@coverage` do?

Reply:
it was my part of remembering which part of the class needed tests :) forgot to remove it.
```suggestion
        $this->query[] = sprintf(
```